src/SqlAsyncCollector.cs

```diff
@@ -171,15 +174,25 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
                 ObjectCache cachedTables = MemoryCache.Default;
                 var tableInfo = cachedTables[cacheKey] as TableInformation;
 
+                int timeout = DEFAULT_TABLE_INFO_CACHE_TIMEOUT_MIN;
+                string timeoutEnvVar = Environment.GetEnvironmentVariable("AZ_FUNC_TABLE_INFO_CACHE_TIMEOUT_MIN");
+                if (!string.IsNullOrEmpty(timeoutEnvVar))
+                {
+                    if (int.TryParse(timeoutEnvVar, NumberStyles.Integer, CultureInfo.InvariantCulture, out timeout))
```

Comment:
If TryParse fails it will set the result to 0 

https://learn.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=net-6.0#system-int32-tryparse(system-string-system-int32@)

So you need to reset it to the default in that case.